src/main/java/com/blackducksoftware/integration/jira/task/maintenance/CleanUpOrphanedTicketsTask.java

```diff
@@ -0,0 +1,225 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package com.blackducksoftware.integration.jira.task.maintenance;
+
+import java.util.List;
+import java.util.Optional;
+import java.util.concurrent.Callable;
+
+import org.apache.commons.lang3.StringUtils;
+import org.apache.log4j.Logger;
+
+import com.atlassian.jira.issue.CustomFieldManager;
+import com.atlassian.jira.issue.Issue;
+import com.atlassian.jira.issue.fields.CustomField;
+import com.atlassian.jira.jql.parser.DefaultJqlQueryParser;
+import com.atlassian.jira.jql.parser.JqlParseException;
+import com.atlassian.jira.jql.parser.JqlQueryParser;
+import com.atlassian.jira.user.ApplicationUser;
+import com.atlassian.jira.workflow.JiraWorkflow;
+import com.atlassian.query.Query;
+import com.blackducksoftware.integration.jira.common.BlackDuckJiraConstants;
+import com.blackducksoftware.integration.jira.common.BlackDuckJiraLogger;
+import com.blackducksoftware.integration.jira.common.JiraUserContext;
+import com.blackducksoftware.integration.jira.common.WorkflowHelper;
+import com.blackducksoftware.integration.jira.common.blackduck.BlackDuckConnectionHelper;
+import com.blackducksoftware.integration.jira.common.exception.JiraIssueException;
+import com.blackducksoftware.integration.jira.common.settings.GlobalConfigurationAccessor;
+import com.blackducksoftware.integration.jira.common.settings.JiraSettingsAccessor;
+import com.blackducksoftware.integration.jira.common.settings.PluginConfigurationAccessor;
+import com.blackducksoftware.integration.jira.common.settings.model.GeneralIssueCreationConfigModel;
+import com.blackducksoftware.integration.jira.common.settings.model.PluginBlackDuckServerConfigModel;
+import com.blackducksoftware.integration.jira.config.JiraServices;
+import com.blackducksoftware.integration.jira.task.issue.handler.JiraIssueServiceWrapper;
+import com.google.common.collect.ImmutableMap;
+import com.google.gson.Gson;
+import com.opensymphony.workflow.loader.ActionDescriptor;
+import com.synopsys.integration.blackduck.exception.BlackDuckApiException;
+import com.synopsys.integration.blackduck.service.BlackDuckService;
+import com.synopsys.integration.blackduck.service.BlackDuckServicesFactory;
+import com.synopsys.integration.exception.IntegrationException;
+import com.synopsys.integration.rest.exception.IntegrationRestException;
+
+public class CleanUpOrphanedTicketsTask implements Callable<String> {
+    private static final String JIRA_QUERY_PARAM_NAME_ISSUE_STATUS = "status";
+    private static final String JIRA_QUERY_PARAM_NAME_ISSUE_TYPE = "issuetype";
+    private static final String JIRA_QUERY_CONJUNCTION = " AND ";
+    private static final String JIRA_QUERY_DISJUNCTION = " OR ";
+
+    // TODO determine reasonable numbers here
+    private static final Integer MAX_BATCH_SIZE = 100;
```

Comment:
Some customers have hundreds of thousands of tickets. This number may not cut it, but I also want to avoid causing a performance hit.